dask/dataframe/io/parquet/core.py

```diff
@@ -373,6 +373,14 @@ def read_parquet(
         "filesystem" key (or "fs" for the "fastparquet" engine) to configure
         the desired file-system backend. However, the top-level ``filesystem``
         argument will always take precedence.
+
+        NOTE: For the "pyarrow" engine, the ``dataset`` options may include a
+        "partitioning" key. However, since ``pyarrow.dataset.Partitioning``
+        objects cannot be serialized, the value can be a dict of key-word
+        arguments for the ``pyarrow.dataset.partitioning`` API
```

Comment:
I wasn't aware that this was a problem. Please open issues on the arrow side about such limitations!

Reply:
Ah, good point. Opened an issue here: https://github.com/apache/arrow/issues/34884

Reply:
Thanks a lot!